codejail/safe_exec.py

```diff
@@ -215,65 +160,50 @@ def filter_unserializable(obj):
 
 def json_safe(d):
     """
-    Return only the JSON-safe part of d.
+    Return only the JSON-safe part of d(a python dict object).
 
     Used to emulate reading data through a serialization straw.
 
     """
 
-    ok_types = (type(None), int, float, str, six.text_type, list, tuple, dict)
+    # six.binary_type is here because bytes are sometimes ok if they represent valid unicode
+    # so we consider them valid for now and try to decode them with decode_object.  If that
+    # doesn't work they'll get dropped later in the process.
+    ok_types = (type(None), int, float, six.binary_type, six.text_type, list, tuple, dict)
 
-    def jsonable(v):
-        if not isinstance(v, ok_types):
-            return False
-        try:
-            json.dumps(v)
-        except Exception:
-            return False
-        return True
- 
-    def filter_unserializable(obj):
+    def decode_object(obj):
+        """
+        Convert an object to be be serialized to json.
+
+        In particulare if we find any byte strings try to convert them to
+        utf-8 strings.
+
+        If for any reason the object contains things that can't be serialized
+        to JSON.  Raise an exception.
```

Comment:
The only exception is due to non-decodable byte strings.  Other non-serializable objects are returned as-is, which is fine because they'll be caught by the dumps/loads later.  Also, these two sentences should be joined with a comma.